src/main/scala/org/ergoplatform/db/LDBFactory.scala

```diff
@@ -0,0 +1,32 @@
+package org.ergoplatform.db
+
+import org.iq80.leveldb.DBFactory
+import scorex.util.ScorexLogging
+
+import scala.util.Try
+
+object LDBFactory extends ScorexLogging {
+
+  private val nativeFactory = "org.fusesource.leveldbjni.JniDBFactory"
+  private val javaFactory   = "org.iq80.leveldb.impl.Iq80DBFactory"
+
+  lazy val factory: DBFactory = {
+    val pairs = for {
```

Comment:
please use .map / .flatmap / .withFilter instead of a for compehension in the main code